velox/dwio/dwrf/test/E2EFilterTest.cpp

```diff
@@ -30,10 +30,38 @@ using dwio::common::MemorySink;
 
 class E2EFilterTest : public E2EFilterTestBase {
  protected:
+  void testWithTypes(
+      const std::string& columns,
+      std::function<void()> customize,
+      bool wrapInStruct,
+      const std::vector<std::string>& filterable,
+      int32_t numCombinations,
+      bool tryNoNulls = false,
+      bool tryNoVInts = false) override {
+    for (int32_t noVInts = 0; noVInts < (tryNoVInts ? 2 : 1); ++noVInts) {
+      useVInts_ = !noVInts;
+      for (int32_t noNulls = 0; noNulls < (tryNoNulls ? 2 : 1); ++noNulls) {
```

Comment:
Feels like `tryNoNulls` should be done in super class?

Reply:
@Yuhta Thanks for reviewing! There're some tests for DWRF, e.g. nullCompactRanges, has tryNoNulls = false, which means it does not test the cases where the data is all not null. But for Parquet, we'd like all tests cover both "With nulls" and "no nulls" cases. So the `tryNoNulls` is for DWRF only.

Reply:
I mean if it is in the function signature, we probably should handle it for all formats where it makes sense (for vints it does not make sense in parquet so we can skip it, we probably should consider removing it from the function signature and add it to the signature only in dwrf tests).

Reply:
Also it makes sense to run `tryNoNulls` in all formats because it is the condition to enable lazy loading on nested structs.